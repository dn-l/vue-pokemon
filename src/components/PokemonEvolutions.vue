<template>
  <div>
    <h2>
      Evolutions
    </h2>
    <div v-if="evolutions">
      <v-list tile>
        <v-list-item
          v-for="evolution in evolutions"
          :key="evolution.id"
          :to="{ name: 'pokemon', params: { id: evolution.id} }"
        >
          <v-list-item-avatar>
            <v-img :src="evolution.image"></v-img>
          </v-list-item-avatar>
          <v-list-item-content>
            <v-list-item-title  v-text="evolution.name" />
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </div>
    <div v-else>
      No evolutions
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  props: ['evolutions', 'requirements'],
});
</script>
