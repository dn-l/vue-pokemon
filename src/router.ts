import Vue from 'vue';
import Router from 'vue-router';
import Pokemons from './views/Pokemons.vue';

export const routes = [
  {
    path: '/',
    name: 'pokemons',
    component: Pokemons,
  },
  {
    path: '/pokemon/:id',
    name: 'pokemon',
    component: () => import(/* webpackChunkName: "pokemon" */ './views/Pokemon.vue'),
  },
  {
    path: '*',
    redirect: '/',
  },
];

Vue.use(Router);

export default new Router({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
});
