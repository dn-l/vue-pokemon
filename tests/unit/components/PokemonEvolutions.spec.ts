import Vue from 'vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import { mount, createLocalVue } from '@vue/test-utils';
import { routes } from '@/router';
import PokemonEvolutions from '@/components/PokemonEvolutions.vue';

const evolutions = [
  {
    id: '1',
    name: 'Ivysaur',
    image: 'https://img.pokemondb.net/artwork/ivysaur.jpg',
  },
  {
    id: '2',
    name: 'Venusaur',
    image: 'https://img.pokemondb.net/artwork/venusaur.jpg',
  },
];

const localVue = createLocalVue();
Vue.use(Vuetify);
localVue.use(Vuex);
localVue.use(VueRouter);

const router = new VueRouter({ routes });

const store = new Vuex.Store({
  getters: {
    isFavourite() {
      return () => false;
    },
  },
});

describe('PokemonEvolutions.vue', () => {
  const wrapper = mount(PokemonEvolutions, {
    propsData: { evolutions },
    store,
    localVue,
    router,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it.each(evolutions)('renders %s', (evolution) => {
    expect(wrapper.text()).toMatch(evolution.name);
  });
});
