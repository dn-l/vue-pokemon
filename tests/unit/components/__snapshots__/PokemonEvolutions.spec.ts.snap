// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonEvolutions.vue matches snapshot 1`] = `
<div>
  <h2>
    Evolutions
  </h2>
  <div>
    <div role="list" class="v-list v-sheet v-sheet--tile theme--light"><a href="#/pokemon/1" class="v-list-item v-list-item--link theme--light" tabindex="0">
        <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
          <div class="v-responsive v-image">
            <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;" name="fade-transition" mode="in-out"></div>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">Ivysaur</div>
        </div>
      </a><a href="#/pokemon/2" class="v-list-item v-list-item--link theme--light" tabindex="0">
        <div class="v-avatar v-list-item__avatar" style="height: 40px; min-width: 40px; width: 40px;">
          <div class="v-responsive v-image">
            <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;" name="fade-transition" mode="in-out"></div>
            <div class="v-responsive__content"></div>
          </div>
        </div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">Venusaur</div>
        </div>
      </a></div>
  </div>
</div>
`;
